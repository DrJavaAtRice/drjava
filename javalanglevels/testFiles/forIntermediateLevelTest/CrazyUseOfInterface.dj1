//This test makes sure that we know that user defined interfaces have the methods defined on Object

class CrazyUseOfInterface {
  MyInterface i;
  
  String iString() {
    return i.toString();
  }
}

interface MyInterface {
}