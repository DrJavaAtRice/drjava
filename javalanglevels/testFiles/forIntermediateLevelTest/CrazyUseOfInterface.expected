// Language Level Converter line number map: dj*->java. Entries: 12
//     1->4         2->5         3->6         4->7         5->8         6->9         7->10        8->11   
//     9->12       10->51       11->52       12->53   
//This test makes sure that we know that user defined interfaces have the methods defined on Object

class CrazyUseOfInterface {
  private final MyInterface i;
  
  public String iString() {
    return i.toString();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public CrazyUseOfInterface(MyInterface i) {
    super();
    this.i = i;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public MyInterface i() {
    return i;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        i() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      CrazyUseOfInterface cast = ((CrazyUseOfInterface) o);
      return (i() != null && i().equals(cast.i()));
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        (i() == null ? 0 : i().hashCode());
  }
}

interface MyInterface {
}
